frontend/packages/db-structure/src/parser/sql/postgresql/index.ts

```diff
@@ -8,12 +9,17 @@ import { processSQLInChunks } from './processSQLInChunks.js'
 export const processor: Processor = async (str: string) => {
   const dbStructure: DBStructure = { tables: {}, relationships: {} }
   const CHUNK_SIZE = 1000
+  const errors: Error[] = []
 
   await processSQLInChunks(str, CHUNK_SIZE, async (chunk) => {
-    const parsed = await parse(chunk)
-    const partial = convertToDBStructure(parsed)
+    const { parse_tree, error } = await parse(chunk)
+    const partial = convertToDBStructure(parse_tree.stmts)
     mergeDBStructures(dbStructure, partial)
+
+    if (error !== null) {
+      errors.push(new UnexpectedTokenWarningError(error.message))
```

Comment:
ask: In this case, are the errors always limited to "UnexpectedToken"? I was wondering if there might be other possibilities.

Reply:
Perhaps that is true, and there are other possibilities.
However, we do not know what errors may occur at present, so we use this approach.
It is assumed that an error class is added together with a test case when another error is known.

Reply:
Thank you, I thought the policy was good. When parsing fails, it can be said that an unexpected token has arrived.